git-config/main.tf

```diff
@@ -0,0 +1,36 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.12"
+    }
+  }
+}
+
+variable "agent_id" {
+    type = string
+    description = "The ID of a Coder agent."
+}
+
+variable "username" {
+  type = string
+  description = "The username of the Coder workspace owner."
+}
+
+variable "user_email" {
+  type = string
+  description = "The email of the Coder workspace owner."
+}
+
+resource "coder_script" "git_config" {
+    agent_id = var.agent_id
+    script = templatefile("${path.module}/run.sh", {
+      CODER_USERNAME = var.username,
+      CODER_EMAIL = var.user_email
+    }) 
+    display_name = "Git Config"
+    icon = "/icon/git.svg" 
+    run_on_start = true
+}
```

Comment:
Can you add two coder parameters to take user_name and email from the user? If they leave it empty, we default to coder deployment username and email.?

With the current approach, Users cannot specify a custom username and email while creating the template. Only the admin can update them.

Reply:
Great call! Adding now.

Reply:
@matifali What if the parameters' values default to the owner email and username, or whatever default is set by the admin? I think this would make the operation more transparent and ergonomic.

Reply:
I've set the default for coder parameters to "default". If the workspace owner wants to populate the fields with thier own credentials they simply replace this value. Let me know if you think this is suitable.

Reply:
I think it works.